consensus/istanbul/core/roundstate.go

```diff
@@ -378,6 +387,31 @@ func (s *roundStateImpl) PreparedCertificate() istanbul.PreparedCertificate {
 	return s.preparedCertificate
 }
 
+func (s *roundStateImpl) SetProposalVerificationStatus(proposalHash common.Hash, verificationStatus error) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.proposalVerificationStatus == nil {
```

Comment:
maybe it's simpler to initialize it on `New()` and then reset it on `StartNewSequence`

Reply:
And on `deserialize`

Reply:
Do you have a strong opinion on doing it that way?

I actually initially was going to do that, but decided to create it on demand, since it would mean only needing to create in one place (as opposed to 3 different places).